src/main/java/com/keniu/models/User.java

```diff
@@ -0,0 +1,63 @@
+package com.keniu.models;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.Filter;
+import org.hibernate.annotations.SQLDelete;
+
+/**
+ * Represents a user in the books shop application.
+ */
+@Entity
+@SQLDelete(sql = "UPDATE user SET is_deleted = true WHERE id=?")
+@Filter(name = "softDeleteFilter", condition = "is_deleted = false")
+@Getter
+@Setter
+@Table(name = "users")
+public class User {
+
+    /** The unique identifier for the user. */
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /** The email address of the user, which should be unique. */
+    @NotBlank
+    @Column(nullable = false, unique = true)
+    @Size(max = 255)
```

Comment:
You already validate the request dto, so these validations are redundant

Reply:
We have already discussed this in another PR. I don't want to repeat myself